rskj-core/src/main/java/co/rsk/trie/TrieStoreImpl.java

```diff
@@ -44,10 +44,29 @@ public TrieStoreImpl(KeyValueDataSource store) {
      */
     @Override
     public void save(Trie trie) {
-        this.store.put(trie.getHash().getBytes(), trie.toMessage());
+        save(trie, true);
+    }
+
+    private void save(Trie trie, boolean forceSaveRoot) {
```

Comment:
I think we need a nice comment here about the meaning of `forceSaveRoot` and why we need it